### PR TITLE
docs: adding User type defintion

### PR DESCRIPTION
**Pull request by avarobinson**
# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
This PR introduces a new `User` type in the Alchemy Signer client and documents its usage.

### Detailed summary
- Added a new `User` type in the `types.ts` file in the signer client package
- Documented the `User` type in the `types.md` file in the resources directory

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: site/resources/types.md [hunk 1]
```diff
@@ -124,6 +124,14 @@ As smart contract accounts are essentially the contract codes that operate on th
 <<< @/../packages/core/src/account/smartContractAccount.ts#SmartContractAccount
 :::
 
+## `User`
+
+`User` is a type that defines the model for the user on a client. It includes identifiers for the user including `email`, `orgId`, `userId`, `userId`, `address` (the corresponding smart contract account address of type Address), and `credentialId`. You can use the [`useUser`](../react/useUser.md) react hook to look up a user.
```
Comment (denniswon) [inline site/resources/types.md:129]: > `(the corresponding smart contract account address of type Address)`

What does this mean? User is the owner/signer EOA, how does this connect to smart contract account? AlchemySigner User can be used something other than the owner of smart contract account(s). A little bit confusing whether this address refers to the EOA address of the Signer/User that maps 1:1 with the user Id/email/credentialId or if this is something else.

Comment (denniswon) [reply site/resources/types.md:129]: Would be great to see clarification on the relationship between User <> Alchemy Signer <> Smart Contract Account (Alchemy Account).

Comment (avarobinson) [reply site/resources/types.md:129]: I actually don't know the answer here. @moldy530 @avasisht23 can you help clarify?

Comment (avarobinson) [reply site/resources/types.md:129]: Pretty positive it is the signer address

File: site/resources/types.md [hunk 2]
```diff
@@ -124,6 +124,14 @@ As smart contract accounts are essentially the contract codes that operate on th
 <<< @/../packages/core/src/account/smartContractAccount.ts#SmartContractAccount
 :::
 
+## `User`
+
+`User` is a type that defines the model for the user on a client. It includes identifiers for the user including `email`, `orgId`, `userId`, `userId`, `address` (the EOA signer address corresponding to the user credentials), and `credentialId`. You can use the [`useUser`](../react/useUser.md) react hook to look up a user.
```
Comment (avasisht23) [inline site/resources/types.md:129]: would clarify this is specific to an Alchemy Signer

Comment (avasisht23) [reply site/resources/types.md:129]:
```suggestion
`User` is a type that defines the model for the details of a user's Embedded Account via an Alchemy Signer. It includes the user's `email`, `orgId`, `userId`, `userId`, `address` (the EOA signer address corresponding to the user credentials), and `credentialId`. You can use the [`useUser`](../react/useUser.md) react hook to look up a user.
```

**Review by denniswon — APPROVED** [review]

**Pull request merged**